src/daemon/daemon.cpp

```diff
@@ -133,13 +146,39 @@ bool t_daemon::run(bool interactive)
       rpc_commands->start_handling(std::bind(&daemonize::t_daemon::stop_p2p, this));
     }
 
+    cryptonote::rpc::DaemonHandler rpc_daemon_handler(mp_internals->core.get(), mp_internals->p2p.get());
+    cryptonote::rpc::ZmqServer zmq_server(rpc_daemon_handler);
+
+    if (!zmq_server.addTCPSocket(zmq_rpc_bind_address, zmq_rpc_bind_port))
+    {
+      LOG_ERROR(std::string("Failed to add TCP Socket (") + zmq_rpc_bind_address
+          + ":" + zmq_rpc_bind_port + ") to ZMQ RPC Server");
+
+      if (interactive)
```

Comment:
looks like that's a local unique_ptr which is being configured before starting the (blocking) p2p.run() call, and that unique_ptr is being set up "if (interactive)".